packages/@lwc/wds-core/src/node/plugins/mock-esm/mutex.js

```diff
@@ -0,0 +1,4 @@
+import { Mutex } from 'async-mutex';
+
+const mutex = new Mutex();
+export default mutex;
```

Comment:
A singleton is probably fine for this. But in case there is a future where multiple instances of the dev server are running in the same Node.js process, it might be better to move it [here](https://github.com/salesforce/lightning-labs/blob/07caaefdc2bb924f268150607921865c06fe90e7/packages/%40lwc/wds-core/src/node/plugins/mock-esm/index.js#L43) and instantiate alongside the `mockedModules` map itself.

Reply:
pushed new changes for this